### PR TITLE
Fix #29266, better error message for deprecated scalar-fill indexed assignment

### LINKED ISSUE #29266: strange error for deprecated scalar-fill indexed assignment
With the deprecation removed, attempting `a[idxs] = scalar` gives an unclear error:

```
julia> a = fill(1, 10);

julia> a[1:5] = 0
ERROR: MethodError: no method matching setindex_shape_check(::Int64, ::Int64)
Closest candidates are:
  setindex_shape_check(::AbstractArray{#s57,1} where #s57, ::Integer) at indices.jl:218
  setindex_shape_check(::AbstractArray{#s57,1} where #s57, ::Integer, ::Integer) at indices.jl:221
  setindex_shape_check(::AbstractArray{#s57,2} where #s57, ::Integer, ::Integer) at indices.jl:225
  ...
Stacktrace:
 [1] macro expansion at ./multidimensional.jl:641 [inlined]
 [2] _unsafe_setindex!(::IndexLinear, ::Array{Int64,1}, ::Int64, ::UnitRange{Int64}) at ./multidimensional.jl:636
 [3] _setindex! at ./multidimensional.jl:631 [inlined]
 [4] setindex!(::Array{Int64,1}, ::Int64, ::UnitRange{Int64}) at ./abstractarray.jl:998
 [5] top-level scope at none:0
```

### PR DESCRIPTION
**Pull request by dourouc05**

### CODE REVIEW

**Comment by mbauman** [comment]
Continuing the discussion from https://github.com/JuliaLang/julia/issues/29266#issuecomment-464202762:

> What would be the next steps?

You have test failures, look into the logs (I find the FreeBSD CI service to have the most readable ones) and see what was wrong.  Or try running `make testall` in your own build of Julia.  In this case it's an easy fix: just do what it says to fix it.

The next step is to add tests for this specific behavior.  Take the failing cases listed in #29266, add the ones you ran into, and put them into a test file — somewhere in `test/broadcast.jl` would make sense for this change.  It doesn't really matter much where you put them within that file — just create a new section there.

**Comment by dourouc05** [comment]
The last commit adds two test cases for this. If the test suite still passes, I guess everything is OK!

**Comment by dourouc05** [comment]
I don't get why the tests don't pass on Windows (and the failure seems unrelated), while they do on FreeBSD…

File: base/indices.jl [hunk 1]
```diff
@@ -231,6 +231,9 @@ function setindex_shape_check(X::AbstractArray{<:Any,2}, i::Integer, j::Integer)
     end
 end
 
+setindex_shape_check(::Any...) =
+    throw(ArgumentError("Scalar assignment to a vector is not supported: maybe you wanted to use `.=`?"))
```
Comment (mbauman) [inline base/indices.jl]: What about this alternative?  Would this make sense?  Or is it simply too pedantic?

```suggestion
    throw(ArgumentError("indexed assignment of a single value to many locations is not supported; perhaps use broadcasting `.=` instead?"))
```

**Comment by dourouc05** [comment]
That sounds better, indeed!

**Comment by dourouc05** [comment]
This PR still has the tag "needs tests", but I provided some (https://github.com/JuliaLang/julia/pull/31085/commits/fe4e503c4e0907b782f33a9ca92ceed76e1fe406). Is there anything else needed from my side for this to get merged?

**Comment by dourouc05** [comment]
This PR still has the tag "needs tests", but I provided some (https://github.com/JuliaLang/julia/pull/31085/commits/fe4e503c4e0907b782f33a9ca92ceed76e1fe406). Is there anything else needed from my side for this to get merged?

**Comment by dourouc05** [comment]
@mbauman Is this PR ready to be merged?

**Review by mbauman — APPROVED** [review]

**Comment by mbauman** [comment]
Thanks for the bump! I'm just going to close and re-open to re-run the tests.  I don't anticipate any trouble, but it's been some time since the last run.  Once they pass, this is good to merge!

**Comment by dourouc05** [comment]
All bots have all tests running perfectly, except analyzegc and Windows, but these failures are unrelated: on Windows, Julia fails to compile; for analyzegc, it seems a rule is missing in the Makefile.

**Pull request merged**